plugins/flatpak/gs-flatpak.c

```diff
@@ -419,6 +484,17 @@ gs_flatpak_add_apps_from_xremote (GsFlatpak *self,
 				  flatpak_remote_get_name (xremote));
 	g_assert (ptask != NULL);
 
+	remote_url = flatpak_remote_get_url (xremote);
+	if (g_str_has_prefix (remote_url, "file://")) {
+		remote_local_repo = g_file_new_for_uri (remote_url);
```

Comment:
Similarly, I’d drop `g_str_has_prefix()` here, hoist the `g_file_new_for_uri()` call, and then check the scheme using `g_file_has_uri_scheme()`.

Reply:
Are you going to do this, or do you disagree?

Reply:
Are there any advantages to doing so?
I'd prefer not to initialize an object just to realize whether it's needed or not (only needed if it's a `file://` scheme). So even though your proposal makes things look a bit more self contained, I don't find it good enough a reason to have the object always allocated (even if the impact is minimal).

I have updated it in the meanwhile to declare the `remote_local_repo` inside the same body where it's used, so it's easier to read and has a more limited scope.

Reply:
Advantages: you don’t have to worry about the difference between `file:`, `file:/`, `file://` and `file:///`; all the string parsing is contained in one place (GLib) which has already been tested.

Reply:
I believe I have seen flatpak checking for `file://` internally, so I don't think we should be covered with the prefix matching, and I prefer this to initializing an object I may not need. So unless I am failing to see the big advantage I actually prefer to keep it like this for now.

Reply:
In that case, please change the prefix match to `file:` rather than `file://`, so that it at least matches the rest of the checks in `flatpak-dir.c`.

Reply:
Okay. That makes sense. Thanks.